utility.py

```diff
@@ -264,3 +267,46 @@ def tag_user(user, tag, channel):
                 queue_message_to_one(messages.tag_message(get_display_name(user_id), tag), channel)
             else:
                 queue_message_to_one(messages.already_tag_message(get_display_name(user_id), tag), channel)
+
+### Banphrase API ###
+
+def check_banphrase(message, channel_name):
+    headers = { 'User-Agent': 'huwobot (https://huwobot.com/)' }
+    params = (('message', message),)
+
+    banphrase_api =  db(opt.CHANNELS).find_one({'name': channel_name})['banphrase_api']
+
+    if not banphrase_api:
+        return False
+
+    try:
+        time.sleep(random.uniform(0.1, 1))
+        response = requests.post('https://' + banphrase_api + '/api/v1/banphrases/test', headers=headers, params=params).json()
+        return response
+    except:
+        raise Exception()
+
+def sanitize_display_name(channel_name, display_name, display_names = None):
+    if display_name != 0:
+        try:
+            return messages.banphrased_name if check_banphrase(display_name, channel_name)['banned'] else display_name
+        except:
+            return messages.banphrase_api_offline
+    elif display_names:
+        display_name_list = []
+        for display_name in display_names:
+            try:
+                display_name_list.append(messages.banphrased_name if check_banphrase(display_name, channel_name)['banned'] else display_name)
+            except:
+                display_name_list.append(messages.banphrase_api_offline)
+        return display_name_list
+
+def sanitize_message(message, channel):
+    banphrase_api_check = check_banphrase(message, channel)
+    if banphrase_api_check and banphrase_api_check['banned']:
+        phrase = banphrase_api_check['banphrase_data']['phrase']
+        if banphrase_api_check['banphrase_data']['case_sensitive']:
+            message = message.replace(phrase)
+        else:
+            message = message.lower().replace(phrase.lower(), messages.banphrased)
```

Comment:
Would this lowercase every message?

Reply:
What's a non-dank way of handling "case_sensitive":true" from pajbot?

Reply:
I see two approaches:
  - the obvious just use case-insensitive regex (just [escape banphrase](https://docs.python.org/3/library/re.html#re.escape))
  - make a copy of the message, find the index(es) of the matching text, replace text from `string_index_of_match` to `string_index_of_match + len(match)` with `[banphrased]` text.